google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/PublisherSnippets.java

```diff
@@ -54,24 +67,112 @@ public void onFailure(Throwable t) {
         System.out.println("failed to publish: " + t);
       }
     });
-    // [END publish]
     return messageIdFuture;
   }
 
   /** Example of creating a {@code Publisher}. */
   // [TARGET newBuilder(TopicName)]
   // [VARIABLE "my_project"]
   // [VARIABLE "my_topic"]
-  public static void newBuilder(String projectName, String topicName) throws Exception {
-    // [START newBuilder]
-    TopicName topic = TopicName.create(projectName, topicName);
+  public static void newBuilder(String projectId, String topicId) throws Exception {
+    TopicName topic = TopicName.create(projectId, topicId);
     Publisher publisher = Publisher.defaultBuilder(topic).build();
     try {
       // ...
     } finally {
       // When finished with the publisher, make sure to shutdown to free up resources.
       publisher.shutdown();
     }
-    // [END newBuilder]
+  }
+
+  public Publisher getPublisherWithCustomBatchSettings(TopicName topicName) throws Exception {
+    // [START publisherBatchSettings]
+    // Batch settings control how the publisher batches messages
+    long requestBytesThreshold = 5000L; // default : 1kb
+    long messageCountBatchSize = 10L; // default : 100
+
+    Duration publishDelayThreshold = Duration.ofMillis(100); // default : 1 ms
+
+    // Batching thresholds have order precedence : request size  > message count > publish delay
```

Comment:
I don't think this is right. We force-send an RPC if the request size grows too large, or we have too many messages, or if we've been waiting too long. I wouldn't call this a precedence: since any of them can force an RPC, they all have the same precedence.